base_product_mass_addition/models/product_mass_addition.py

```diff
@@ -0,0 +1,96 @@
+# © 2019 Today Akretion
+# @author Pierrick Brun <pierrick.brun@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+import logging
+
+from odoo import api, models
+
+_logger = logging.getLogger(__name__)
+
+
+class ProductMassAddition(models.AbstractModel):
+    _name = "product.mass.addition"
+    _description = "inherit this to add a mass product addition function\
+                    to your model"
+
+    @api.model
+    def _common_action_keys(self):
+        """Call it in your own child module"""
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": "product.product",
+            "target": "current",
+            "context": {"parent_id": self.id, "parent_model": self._name},
+            "view_mode": "tree",
+        }
+
+    def _prepare_quick_line(self, product):
+        res = self._get_quick_line_qty_vals(product)
+        res.update({"product_id": product.id})
+        return res
+
+    def _get_quick_line(self, product):
+        raise NotImplementedError
+
+    def _add_quick_line(self, product, lines_key=""):
+        if not lines_key:
+            raise NotImplementedError
+        vals = self._prepare_quick_line(product)
+        vals = self._complete_quick_line_vals(vals)
+        self.write({lines_key: [(0, 0, vals)]})
+
+    def _update_quick_line(self, product, line):
+        if product.qty_to_process:
+            # apply the on change to update price unit if depends on qty
+            vals = self._get_quick_line_qty_vals(product)
+            vals["id"] = line.id
+            vals = self._complete_quick_line_vals(vals)
+            line.write(vals)
+        else:
+            line.unlink()
+
+    def _get_quick_line_qty_vals(self, product):
+        raise NotImplementedError
+
+    def _complete_quick_line_vals(self, vals, lines_key=""):
+        if not lines_key:
+            raise NotImplementedError
+        init_keys = ["product_id"]
+        init_vals = {key: val for key, val in vals.items() if key in init_keys}
+        update_vals = {key: val for key, val in vals.items() if key not in init_keys}
+        # Try to use onchange_helper if installed
+        try:
```

Comment:
I think you should better test if the module ``onchange_helper`` is installed. So, you avoid a try catch that could hide other erros.

Reply:
That would require to do a `.search()` everytime this function is called, I'm not sure this is a good idea to add that (or there is a less costly way to do it)

Also, I only catch AttributeError, which shouldn't happen often outside of development work and I logged the error message to make it more obvious